uitools/import/Esri/ArcGISRuntime/Toolkit/Controller/BookmarksViewController.qml

```diff
@@ -0,0 +1,110 @@
+/*******************************************************************************
+ *  Copyright 2012-2022 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+import QtQml 2.15
+import Esri.ArcGISRuntime 100.15
+import QtQml.Models 2.15
+
+/*!
+   \qmltype BookmarkController
+   \inqmlmodule Esri.ArcGISRuntime.Toolkit
+   \since Esri.ArcGISRuntime 100.15
+   \ingroup ArcGISQtToolkitUiQmlControllers
+   \brief The controller part of a BookmarksView. This class handles the
+   management of the Bookmark objects and zooming onto the bookmarks' view extents.
+ */
+
+QtObject {
+    id: controller
+    /*!
+      \qmlproperty GeoView geoView
+      \brief The geoview of the controller.
+     */
+    property GeoView geoView: null;
+
+    /*!
+      \qmlproperty ListModel bookmarks
+      \brief The bookmarks of Bookmark list objects.
+
+      Internally, the list is a QML ListModel containing elements of type ListElement.
+      Each ListElement has a `modelData` and property `name`, which maps to a Bookmark.
+     */
+    readonly property alias bookmarks: internal.bookmarks
+
+    /*!
+       \brief Method that takes a Bookmark \a modelData and  zoom to that bookmark in the `geoView`.
+     */
+    function zoomToBookmarkExtent(modelData) {
+        geoView.setBookmark(modelData);
+    }
+
+    /*! \internal */
+    property QtObject internal : QtObject {
+        id: internal
+
+        // qmlproperty Bookmark rawBookmarks
+        // The bookmarks, stored in the geoView.
+        // This property can experience changes, which are copied into the internal `bookmarks` ListModel.
```

Comment:
assuming these are meant to be internal? They aren't enclosed in qdoc tags, just comments

Reply:
Yes, these are internal comments only - they are not to be published in the published documentation. They should, however, be present here in the code as comments. The formatting (i.e. wording) of these comments are the same as the docs, because that's a useful/easy-to-read formatting.